script.js

```diff
@@ -0,0 +1,186 @@
+          Buffer.from(content, encoding).toString('utf-8')
+      );
+
+      /** @type {import('yaml').YAMLMap<string>} */
+      // @ts-expect-error Why is this `unknown`?
+      const jobs = yamlDocument.get('jobs');
+
+      for (const { value: job, key: jobName } of jobs.items) {
+        /** @type {import('yaml').YAMLSeq<number> | undefined} */
+        const nodeVersions = job
+            .get('strategy')
+            ?.get('matrix')
+            ?.get('node_version');
+
+        if (nodeVersions) {
+          yamlDocument.setIn(['jobs', jobName, 'strategy', 'matrix', 'node_version'], NODE_VERSIONS);
+        }
+      }
+
+      return prettier.format(stringify(yamlDocument), { parser: 'yaml' });
+    }
+    octokit.log.info(
+        'The repository \'%s\' does not have any usage of node_version %s to be removed in its GitHub Actions',
+        repo,
+        `${NODE_VERSIONS_TO_REMOVE}`
+    );
+
+    return content;
+  }
+
+  /** @type {Exclude<import('@octokit/openapi-types').components["schemas"]["content-tree"]['entries'], undefined> | [import('@octokit/openapi-types').components["schemas"]["content-tree"]]} */
+  let files;
+
+  // Get all files from .github/workflows folder
+  try {
+    /** @type {import('@octokit/types').OctokitResponse<import('@octokit/openapi-types').components["schemas"]["content-tree"]>} */
+    // @ts-ignore Overriding the type of the response for the correct type with the `object` media type
+    const { data } = await octokit.request(
+        'GET /repos/{owner}/{repo}/contents/{path}',
+        {
+          owner,
+          repo,
+          path: PATH,
+          mediaType: {
+            format: 'object'
+          }
+        }
+    );
+
+    // We know that the path is a directory, we do this check to appease typescript and get rid of the `undefined`
+    if (Array.isArray(data.entries)) {
+      files = data.entries.filter(entry => isYamlFile(entry.name));
+    } else {
+      files = [data];
+    }
+  } catch (e) {
+    if (e.status === 404) {
+      octokit.log.warn(`"${PATH}" path not found in ${repository.full_name}`);
+
+      return;
+    }
+    throw e;
+  }
+
+  if (files.length) {
+    for (const file of files) {
+      changes.push({
+        files: {
+          [`.github/workflows/${file.name}`]: updateYamlFile.bind(null, file)
+        },
+        commit: `ci: stop testing against NodeJS ${NODE_VERSIONS_STRING}`,
+        emptyCommit: false
+      });
+    }
+  } else {
+    octokit.log.info('There is no file %s in repository %s', PATH, repo);
+  }
+
+  //
+  // Pull Request
+  //
+
+  const pr = await composeCreatePullRequest(octokit, {
+    owner,
+    repo,
+    title: `ci: stop testing against NodeJS ${NODE_VERSIONS_STRING}`,
+    body: `BREAKING CHANGE: Drop support for NodeJS ${NODE_VERSIONS_STRING}`,
+    head: branchName,
+    changes,
+    createWhenEmpty: false,
+    update: true
+  });
+
+
+  if (pr)
+    octokit.log.info(`Pull request created: ${pr.data.html_url}`);
```

Comment:
Should we wrap this in a try-catch and `ocotkit.log.error` accordingly or do we want to just throw?

Reply:
I don't think it's warranted